text/src/autogluon/text/automm/data/trivial_augmenter.py

```diff
@@ -0,0 +1,252 @@
+"""
+This file implements TrivialAugment.(https://arxiv.org/abs/2103.10158) We extend it for multi-modality setting. 
+
+Code is partically adapted from its official implementation https://github.com/automl/trivialaugment
+"""
+
+import random
+from selectors import EpollSelector
+from PIL import ImageOps, ImageEnhance, ImageFilter, Image, ImageDraw
+import nlpaug.augmenter.word as naw
+from .utils import InsertPunctuation
+import nltk
+
+
+def scale_parameter(level, maxval, type):
+    """Helper function to scale `val` between 0 and maxval .
+    Parameters
+    ----------
+    level: Level of the operation that will be between [0, image_parameter_max].
+    maxval: Maximum value that the operation can have.
+    type: return float or int
+
+    Returns
+    -------
+    An adjust scale
+    """
+    if type == "float":
+        return float(level) * maxval / image_parameter_max
+    elif type == "int":
+        return int(level * maxval / image_parameter_max)
+
+
+class TransformT(object):
+    """Each instance of this class represents a specific transform."""
+
+    def __init__(self, name, xform_fn):
+        self.name = name
+        self.xform = xform_fn
+
+    def __repr__(self):
+        return "<" + self.name + ">"
+
+    def augment(self, level, data):
+        return self.xform(data, level)
+
+
+identity = TransformT("identity", lambda data, level: data)
+
+
+auto_contrast = TransformT("AutoContrast", lambda pil_img, level: ImageOps.autocontrast(pil_img))
+
+
+equalize = TransformT("Equalize", lambda pil_img, level: ImageOps.equalize(pil_img))
+
+
+def _rotate_impl(pil_img, level):
+    """Rotates `pil_img` from -30 to 30 degrees depending on `level`."""
+    max = 30
```

Comment:
To make these functions more general, we'd better no hard-code the `min` and `max` values inside specific functions. Instead, consider defining them inside the `set_image_augmentation_space`.

Reply:
Making the function start with “_” should be fine. People know that it’s internal.

Reply:
I think if user want to do rotate as augmentation, they can specify under train_transform _type and then use pytorch Transform. The operations in this file are for trivial augment algorithm, and it is the parameters used by the paper.